src/base/memory_alloc.h

```diff
@@ -291,4 +296,53 @@ enum
   (PRIVATE_MALLOC_HEADER *)((char *)(ptr) - PRIVATE_MALLOC_HEADER_ALIGNED_SIZE)
 #endif /* SA_MODE */
 
+#if defined (__cplusplus)
+
+class PRIVATE_UNIQUE_PTR_DELETER
+{
+public:
+  PRIVATE_UNIQUE_PTR_DELETER ():thread_p (NULL)
```

Comment:
you can put PRIVATE_UNIQUE_PTR_DELETER (THREAD_ENTRY * thread_p = NULL) for default argument and remove default constructor